api/envoy/service/ext_proc/v3/external_processor.proto

```diff
@@ -332,3 +344,24 @@ message BodyMutation {
     bool clear_body = 2;
   }
 }
+
+// This message specifies a new timer duration and the original
+// request direction. If these fields are not configured correctly,
+// the response message will be ignored by the Envoy ext_proc filter.
+message NewProcessingTimeout {
+  // New timeout range in milliseconds.
+  enum NewTimeoutRange {
```

Comment:
Doesn't look like this enum is actually used anywhere.  Is this actually needed?

Reply:
Thanks for the comments.

The enum is used in source/extensions/filters/http/ext_proc/ext_proc.cc. All the validation logic is added there.

Reply:
It doesn't look like these are used in the API itself, so I don't think the enum should be defined here in the proto file.

Reply:
Yeah, I thought about this.

I can just add the restrictions in the proto file comments. Then define the constants in the ext_proc.cc and enforce the restriction there. But then I am concerning how to synchronize them?  For example if later on we want to change that restrictions in the proto as control plane has new requirements, how do people know to change the corresponding implementations in ext_proc.cc?

Reply:
I don't think it's necessary nor desirable to forcibly encode these constraints into the API.  The xDS API is not specific to Envoy; there may be other implementations of this feature in other data planes in the future (e.g., gRPC).  And it's possible that some implementations will intentionally allow a broader range of values here.  All you can really do is document what you think makes sense, and others can update it as the world changes.

Reply:
Done